File: sdk/fsharp/src/HiTechCloudSDK/CloudGpu.fs
namespace HiTechCloud.SDK

/// Cloud GPU: GPU instance management
type CloudGpuResource(httpClient: HttpClient) =

    /// PUT /api/service/{id}/vms/reboot - Reboot virtual server
    member this.Reboot() =
        async {
            return! httpClient.putAsync("/api/service/{id}/vms/reboot")
        }

    /// PUT /api/service/{id}/vms/stop - Stop virtual server
    member this.Stop() =
        async {
            return! httpClient.putAsync("/api/service/{id}/vms/stop")
        }

    /// PUT /api/service/{id}/vms/start - Start virtual server
    member this.Start() =
        async {
            return! httpClient.putAsync("/api/service/{id}/vms/start")
        }

    /// GET /api/service/{id}/vms/firewall - Get firewall rules
    member this.GetFirewallRules() =
        async {
            return! httpClient.getAsync("/api/service/{id}/vms/firewall")
        }

    /// POST /api/service/{id}/vms/firewall - Add firewall rules
    member this.AddFirewallRules() =
        async {
            return! httpClient.postAsync("/api/service/{id}/vms/firewall")
        }

    /// DELETE /api/service/{id}/vms/firewall/{position} - Remove firewall rule
    member this.RemoveFirewallRule() =
        async {
            return! httpClient.deleteAsync("/api/service/{id}/vms/firewall/{position}")
        }

File: sdk/clojure/src/hitechcloud/sdk/cloud_gpu.clj
(ns hitechcloud.sdk.cloud_gpu
  (:require [hitechcloud.sdk.http-client :as http]))

;; Cloud GPU: GPU instance management

(defn reboot "PUT /api/service/{id}/vms/reboot - Reboot virtual server" [client & [params]] (http-put client "/api/service/{id}/vms/reboot" params))

(defn stop "PUT /api/service/{id}/vms/stop - Stop virtual server" [client & [params]] (http-put client "/api/service/{id}/vms/stop" params))

(defn start "PUT /api/service/{id}/vms/start - Start virtual server" [client & [params]] (http-put client "/api/service/{id}/vms/start" params))

(defn get_firewall_rules "GET /api/service/{id}/vms/firewall - Get firewall rules" [client & [params]] (http-get client "/api/service/{id}/vms/firewall" params))

(defn add_firewall_rules "POST /api/service/{id}/vms/firewall - Add firewall rules" [client & [params]] (http-post client "/api/service/{id}/vms/firewall" params))

(defn remove_firewall_rule "DELETE /api/service/{id}/vms/firewall/{position} - Remove firewall rule" [client & [params]] (http-delete client "/api/service/{id}/vms/firewall/{position}" params))

